@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -910,41 +944,45 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
                             vertices[i + 1],
                             vertices[i + 2] + minHeight * scaleFactor
                         );
+                        if (texCoordType !== undefined) {
+                            textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                        }
+
                         if (isExtruded) {
                             positions.push(
                                 vertices[i],
                                 vertices[i + 1],
                                 vertices[i + 2] + height * scaleFactor
                             );
-                        } else if (isTextured) {
-                            textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                            if (texCoordType !== undefined) {
+                                textureCoordinates.push(vertices[i + 3], vertices[i + 4]);
+                            }
                         }
                     }
 
                     // Add the footprint/roof indices to the index buffer.
                     const vertexStride = isExtruded ? 2 : 1;
                     for (let i = 0; i < triangles.length; i += 3) {
-                        const v0 = baseVertex + triangles[i + 0] * vertexStride;
-                        const v1 = baseVertex + triangles[i + 1] * vertexStride;
-                        const v2 = baseVertex + triangles[i + 2] * vertexStride;
-                        indices.push(v0, v1, v2);
-
                         if (isExtruded) {
+                            //Take every second vertex.
                             const v3 = baseVertex + triangles[i + 0] * vertexStride + 1;
                             const v4 = baseVertex + triangles[i + 1] * vertexStride + 1;
                             const v5 = baseVertex + triangles[i + 2] * vertexStride + 1;
                             indices.push(v3, v4, v5);
+                        } else {
+                            const v0 = baseVertex + triangles[i + 0] * vertexStride;
```

Comment:
we know that if `isExtruded` is false, that `vertexStride` is `1`, so I think this can be removed.

Reply:
no, vertex stride is either 3 or 5 if we have texture coordinates.

Reply:
`const vertexStride = isExtruded ? 2 : 1;`

`vertexStride` is therefore `1` in the else condition.

Reply:
even better, you don't need the if condition / else condition at all.